staging/src/k8s.io/client-go/util/workqueue/delaying_queue_test.go

```diff
@@ -230,7 +475,7 @@ func BenchmarkDelayingQueue_AddAfter(b *testing.B) {
 }
 
 func waitForAdded(q DelayingInterface, depth int) error {
-	return wait.Poll(1*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+	return wait.Poll(1*time.Millisecond, 5*time.Second, func() (done bool, err error) {
```

Comment:
are these changes necessary?

Reply:
The new functionality tests slow the testing down considerably so my question was more along the lines of "is that change an issue"?  i.e. do we need a long as 10 seconds to confirm this state?  I can revert it - will just make the tests really long.

Reply:
Keeping the tests as short as possible is a big quality of life improvement. I like to use fake clocks if possible.

Reply:
Using real clocks / real time.Sleep etc tends to make things racy.

Reply:
ok, I had a good look at the tests and found a nasty race due to the possibility of moving the fake clock forward in-between an item being added to the delayed queue and the new NextReadyAt time being calculated.  I've had to go back and look at the waitingLoop code and refactor it slightly.  I think the resultant code is fairly similar to what was there before but with the calculate pulled out to its own method so that it can be performed appropriately when popping items off the queue but also when making changes caused by inbound entries.